package/run.sh

```diff
@@ -20,7 +20,33 @@ else
     FORCE_RESTART=false
 fi
 
-env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+
+# work around for https://github.com/k3s-io/k3s/issues/2306
+# depends on journald being available on first boot, so this may not work on all providers (e.g. Digital Ocean)
+DELAY=10
+while ! env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
+do
+    # We need to get the right systemd service based off of the role we have, $INSTALL_K3S_EXEC will be set if we are a worker only node, in which case the systemd service is k3s-agent.service
+    SERVICE=k3s-agent.service
```

Comment:
If the goal is to only look for the "etcd join failed", the k3s-agent should not ever emit this message as only "server" nodes are etcd.